library/core/src/str/pattern.rs

```diff
@@ -601,6 +608,17 @@ where
     }
 }
 
+impl<const N: usize> MultiCharEq for &[char; N] {
```

Comment:
This might be a silly question, but why `for &[char; N]` instead of `for [char; N]`?

Reply:
Actually that's what I started with, but then I figured this is more flexible/performant for large arrays. Definitely open to discussion on that. Maybe add both?

Reply:
I think you do want the reference to solve those bugs where one intends to write a literal `&[char]` slice pattern, but it's really `&[char; N]`.

Reply:
I think both -- `['o', 'l']` for the simple case, and `&[...]` for the case cuviper mentions.

(I might just delegate both to the impl for slices, though, if there's an easy way to do that.)

Reply:
Okay, added both. Unfortunately lifetimes force a lot of separate impls to satisfy `into_searcher`.